src/compile/data/formatparse.ts

```diff
@@ -141,12 +147,12 @@ export class ParseNode extends DataFlowNode {
    * Creates a parse node from "explicit" parse and "implicit" parse and updates ancestorParse.
    */
   private static makeWithAncestors(parent: DataFlowNode, explicit: Dict<string>, implicit: Dict<string>, ancestorParse: AncestorParse) {
-    // We should not parse what has already been parsed in a parent (explicitly or implicitly) or what has been derived (maked as "derived").
+    // We should not parse what has already been parsed in a parent (explicitly or implicitly) or what has been derived (maked as "derived") and we don't need to flatten again.
     for (const field of keys(implicit)) {
       const parsedAs = ancestorParse.getWithExplicit(field);
       if (parsedAs.value !== undefined) {
         // We always ignore derived fields even if they are implicitly defined because we expect users to create the right types.
-        if (parsedAs.explicit || parsedAs.value === implicit[field] || parsedAs.value === 'derived') {
+        if (parsedAs.explicit || parsedAs.value === implicit[field] || parsedAs.value === 'derived' || implicit[field] === 'flatten') {
```

Comment:
Why do you need to delete flatten?

Reply:
If we already parse a field, there is no need to flatten (flatten is just using a calculate with no change to the data).

Reply:
> If we already parse a field

Isn't this checked from `parseAs` -- why is it related to the new `|| implicit[field] === 'flatten'`?

Reply:
The field was parsed before. So now the question is, do we parse it again as something different (e.g. string to number). If the new parse is flatten, we don't need to do anything.